cpp/oneapi/dal/backend/primitives/reducer.cpp

```diff
@@ -0,0 +1,221 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <algorithm>
+#include <cmath>
+#include <exception>
+#include <limits>
+
+#ifdef ONEAPI_DAL_DATA_PARALLEL
+    #include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/dal/backend/primitives/common.hpp"
+#include "oneapi/dal/backend/primitives/reducer.hpp"
+#include "oneapi/dal/data/array.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEAPI_DAL_DATA_PARALLEL
+
+namespace impl {
+
+template <typename UnaryFunctor, typename BinaryFunctor, typename Float, bool VectorsAreRows>
+struct reducer_singlepass_kernel {
+public:
+    const BinaryFunctor binary;
+    const UnaryFunctor unary;
+
+private:
+    typedef cl::sycl::
+        accessor<Float, 1, cl::sycl::access::mode::read_write, cl::sycl::access::target::local>
+            local_acc_t;
```

Comment:
use `using`. You can avoid typing `cl::sycl::something` and make it just `sycl::something`